dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -221,249 +241,260 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
+
+                        builder.Name(backgroundTaskFullName);
 
-                        if (std::wstring_view(name).substr(0, backgroundTaskName.size()) != backgroundTaskName)
-                        {
-                            return false;
-                        }
+                        PushNotificationTrigger trigger{};
+                        builder.SetTrigger(trigger);
 
-                        if (name == backgroundTaskFullName)
-                        {
-                            s_pushTriggerRegistration = task.Value();
-                            return true;
-                        }
+                        auto builder5{ builder.as<winrt::IBackgroundTaskBuilder5>() };
+                        builder5.SetTaskEntryPointClsid(m_registeredClsid);
+ 
+                        m_pushTriggerRegistration = builder.Register();
+                    }
+                }
 
-                        throw winrt::hresult_invalid_argument(L"RegisterActivator has different clsid registered.");
-                    }) };
+                // Register a sink for the application to receive foreground raw notifications
+                RegisterForegroundSinkHelper();
 
-                if (!isTaskRegistered)
+                // Register a PushNotificationBackgroundTask to handle background activation scenarios
+                if (!m_comActivatorRegistration)
                 {
-                    builder = BackgroundTaskBuilder();
-                    builder.Name(backgroundTaskFullName);
-
-                    PushNotificationTrigger trigger{};
-                    builder.SetTrigger(trigger);
+                    THROW_IF_FAILED(::CoRegisterClassObject(
+                        m_registeredClsid,
+                        winrt::make<PushNotificationBackgroundTaskFactory>().get(),
+                        CLSCTX_LOCAL_SERVER,
+                        REGCLS_MULTIPLEUSE,
+                        &m_comActivatorRegistration));
+                }
 
-                    THROW_HR_IF(E_NOTIMPL, !AppModel::Identity::IsPackagedProcess());
+                scopeExitToCleanRegistrations.release();
+            }
+            else
+            {
+                auto scopeExitToCleanRegistrations{ wil::scope_exit([&]()
+                {
+                    m_comActivatorRegistration.reset();
+                    m_protocolRegistration = false;
 
-                    // In case the interface is not supported, let it throw.
-                    auto builder5{ builder.as<winrt::IBackgroundTaskBuilder5>() };
-                    builder5.SetTaskEntryPointClsid(taskClsid);
-                    winrt::com_array<winrt::IBackgroundCondition> conditions = details.GetConditions();
-                    for (auto condition : conditions)
+                    auto notificationsLongRunningPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+                    if (PushNotificationHelpers::IsPackagedAppScenario())
```

Comment:
This check doesn't line up with the if condition which presumes that this is UnpackagedAppScenario. What am i missing here?

Reply:
Ooops. Changing this to IsPackagedProcess